StratisCore.UI/src/app/shared/services/signalr-service.ts

```diff
@@ -0,0 +1,97 @@
+import { EventEmitter, Injectable } from '@angular/core';
+import "@aspnet/signalr";
+import * as signalR from '@aspnet/signalr';
+import { HttpClient } from "@angular/common/http";
+import { ISignalRService } from "@shared/services/interfaces/services.i";
+import { GlobalService } from "@shared/services/global.service";
+import { RestApi } from "@shared/services/rest-api";
+import { ErrorService } from "@shared/services/error-service";
+
+export interface SignalRConnectionInfo {
+  signalRUri: string;
+  signalRPort: string;
+}
+
+export interface SignalRMessageHandler {
+  messageType: string;
+  onEventMessageDelegate: (message: any) => void
+}
+
+@Injectable({
+  providedIn: "root"
+})
+export class SignalRService extends RestApi implements ISignalRService {
+  private connection: signalR.HubConnection;
+  private onMessageReceivedHandlers: Array<SignalRMessageHandler> = [];
+
+  constructor(http: HttpClient,
+              globalService: GlobalService,
+              errorService: ErrorService) {
+    super(globalService, http, errorService);
+
+    // TODO: consider multiple Hub support
+    this.connect("events");
+  }
+
+  public registerOnMessageEventHandler<TMessage>(messageType: string, onEventMessageReceivedHandler: (message: TMessage) => void): void {
+    this.onMessageReceivedHandlers.push({
+      messageType: messageType,
+      onEventMessageDelegate: onEventMessageReceivedHandler
+    });
+  }
+
+  public onConnectionFailed: EventEmitter<Error> = new EventEmitter<Error>();
+
+  public connect(hubName: string): void {
```

Comment:
Do we need any connection recovery logic, for example if connection drops for any reason, show we try to reconnect?

Reply:
I have add this yesterday